### DISCUSSION TITLE
Issue updating select field with $app.dao()

### DISCUSSION THREAD

**Discussion by natecasp** [pocketbase/pocketbase#4403] [Q&A] [answered]
I am trying to use the javascript engine to update a select field whenever a record is changed.
I have successfully used the .set(key, value) with other fields in the past, but am having trouble setting the value of a select field with an array of values.

My select field is set up to allow multiple selections.

Here is the jist of my code:
`//Get the record from collection "orders"`
`const record = $app.dao().findRecordById("orders", id);`
`//Access the record model`
`var expanded = JSON.parse(JSON.stringify(record));`
`//Get the current selections for the 'sent' select field (this returns an array)`
`var oldsettings = expanded.sent;`
`//Add the new value to the settings`
`var newsettings = oldsettings.push(newvalue);`
`//Update the record and save it`
`record.set("sent", newsettings);`
`$app.dao().saveRecord(record)`

Currently I am not getting any errors, but the record does not get updated with the added new selections.
Is there an issue with the formatting of the array?
Using postman to send an http patch request with the same array the record updates successfully.

Thanks!

**Comment by ganigeorgiev** [comment]
The [array `push()` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push#return_value) appends a new value to the existing array and doesn't returns a new one (_the return value will be the length of the new array_).

Or in other words, the `oldSettings` variable will be updated with your new value. Here is a slightly modified version of your example:

```js
const record = $app.dao().findRecordById("orders", id)

const sent = record.get("sent")
sent.push(newvalue)

record.set("sent", sent)

$app.dao().saveRecord(record)
```

**Reply by natecasp** [reply]
@ganigeorgiev thanks for the quick response.

That makes sense to me, I updated my code to reflect your changes.
When I console.log(sent) after sent.push(newvalue), I can see that it returns an array with the old values and the new values which is the desired effect, but the record still is not updated after the set and saveRecord operations.

**Reply by ganigeorgiev** [reply]
I've just tested it locally and the value is updated successfully for me (for a multiple `select` field).

If you are still not able to resolve it, please provide a more complete code sample and export your collections configuration as json (_Admin UI > Settings > Export collections_).

**Reply by natecasp** [reply, marked as answer]
My mistake, I just realized that I am trying to do this in a onRecordBeforeUpdateRequest, in which case I just need to use e.record.set("sent", sent). This updates successfully.
